helpers/types.rs

```diff
@@ -55,3 +55,27 @@ macro_rules! construct_from {
 }
 
 construct_from!(u8, u16, u32, u64, usize, i8, i16, i32, i64);
+
+// Fee size that is charged for trading.
+// Denominated in parts per 1000.
+// ie 1 = 0.1%, 10 = 1%, 100 = 10%
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+#[repr(u8)]
+pub enum FeeSize {
```

Comment:
Why do we need this `FeeSize`? It's giving less flexibility. I think using `FeeMillis` (just u32) in place of `FeeSize` would be OK.

Reply:
The reasoning was that if we allow for setting it as `u32` then we need some validation logic. The `fee` would have to be smaller than `1000` (b/c of the formula in which this variable is used in). So the constructor would have be change its type (to return `Result<Self, Error>`). Also, right now the smallest fee possible is `1` which translates to `0.1%`. This isn't great either b/c some pools might want even smaller fee - like `0.05%`, or `0.01%` . To accommodate for that we'd have to rewrite more of the code in the `Pair` since we have multiplied by 1000 (it's one of the constants). So, even more _list minute changes_.

I figured all of that is not worth the additional complexity since this AMM model does not work well for pools with low fees in general. It's susceptible to higher slippage (unlike Curve's model) so the fees _should_ be higher to offset that risk. I thought that since it's just a temporary solution, until we have implemented the RateSwap from Curve (and we're already working on it), and/or UniswapV3 style (or Invariant), it may not be worth all of the troubles.